### PR TITLE
Add saving and loading from URL hash

### PR DESCRIPTION
**Pull request by cemulate**
This addresses issue #8 in a slightly different way.

Instead of saving and loading JSON (just another type of markup along with TikZ), this adds a button that encodes the current diagram and puts it in the URL hash, copying the resulting URL to the clipboard at the same time. This allows the same diagram to be loaded by navigating to the copied URL.

The new button is next to the TikZ button:
image

and works essentially the same way.

### CODE REVIEW

File: img/tools/link.svg [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="iso-8859-1"?>
```
Comment (yishn) [inline img/tools/link.svg]: Can you pass this SVG into something like [SVGOMG](https://jakearchibald.github.io/svgomg/) to remove all the unnecessary stuff such as these empty `<g>` elements?

File: src/components/App.js [hunk 2]
```diff
@@ -80,6 +89,29 @@ export default class App extends Component {
         }
     }
 
+    getLink = (event) => {
```
Comment (yishn) [inline src/components/App.js]: Can you rename this function to `copyLink` for consistency with `copyCode`?

File: src/components/App.js [hunk 3]
```diff
@@ -80,6 +89,29 @@ export default class App extends Component {
...
+
+        // This prevents the href="#" on this button from causing a hash change after we set the hash with replaceState below
+        event.preventDefault();
```
Comment (yishn) [inline src/components/App.js]: I've added `evt.preventDefault()` to the `<Button>` component in master, so we can omit this line.

File: src/components/App.js [hunk 4]
```diff
@@ -80,6 +89,29 @@ export default class App extends Component {
...
+
+        if (this.state.confirmLink) return
```
Comment (yishn) [inline src/components/App.js]: Can you add `confirmLink` to the initial state in the constructor?

File: img/tools/link.svg (hunk 1 above)
Comment (cemulate) [reply img/tools/link.svg]: Great tool; thanks for informing me about it!

**Comment by cemulate** [comment]
So, I tried to do some dark git magic to rebase my branch so I could have your fix from `master`. But, I've never done that in conjunction with a fork and a pull request, so I wasn't sure how this would turn out. The only odd thing is that your commit on master is included in this branch -- but once the merge happens I think it'll just show up as one.

**Comment by yishn** [comment]
Thank you so much!

**Comment by cemulate** [comment]
No problem - thanks for the extremely helpful tool!

**Pull request merged**